contrib/win32/win32compat/shell-host.c

```diff
@@ -207,32 +207,48 @@ SendKeyStroke(HANDLE hInput, int keyStroke, char character)
 	ir.Event.KeyEvent.wVirtualKeyCode = keyStroke;
 	ir.Event.KeyEvent.wVirtualScanCode = 0;
 	ir.Event.KeyEvent.dwControlKeyState = 0;
-	ir.Event.KeyEvent.uChar.UnicodeChar = 0;
-	ir.Event.KeyEvent.uChar.AsciiChar = character;
+	ir.Event.KeyEvent.uChar.UnicodeChar = character;
 
-	WriteConsoleInputA(hInput, &ir, 1, &wr);
+	WriteConsoleInputW(hInput, &ir, 1, &wr);
 
 	ir.Event.KeyEvent.bKeyDown = FALSE;
-	WriteConsoleInputA(hInput, &ir, 1, &wr);
+	WriteConsoleInputW(hInput, &ir, 1, &wr);
+}
+
+void
+initialize_keylen()
+{
+	for(int i = 0; i < ARRAYSIZE(keys); i++)
+		keys[i].in_key_len = (int) wcslen(keys[i].in);
 }
 
 void 
 ProcessIncomingKeys(char * ansikey)
 {
-	int keylen = (int) strlen(ansikey);
-
-	if (!keylen)
-		return;
+	wchar_t *buf = utf8_to_utf16(ansikey);
+		
+	if (!buf) {
```

Comment:
fatal is not available in shell-host.c and also printf() output is displayed on client side..